### DISCUSSION TITLE
Weird (insecure maybe?) behaviour when resetting password

### DISCUSSION THREAD

**Discussion by tawjaw** [supabase/supabase#4773] [Questions] [answered]
Hello 

I have two different accounts. In browser1 account1 is logged in. in browser2 account2 requested forgot password. 

I got the email to reset the password for account2 and clicked on the link, it opened in browser1 (the default browser that account1 is logged in using). 

And I changed the password **for account1 which was logged in**!

account1 did not confirm with email to changing the password. The reset password was sent to account2 email. 

is this normal behaviour? I find this a security problem.

**Comment by awalias** [comment, marked as answer]
thanks for the question @tawjaw !

The forgotten password link is actually a form of 'magic link' that logs the user in with a reset code, it sounds like what might have happened here is:

1. link in email is clicked, opening browser1
2. reset code is exchanged with auth back end, issuing an access_token for account2
3. user is forwarded to your app with the new access_token as a URI fragment

at this point I'm not sure exactly how supabase-js behaves:
- does it ignore the new access_token for account2, given that it already has an active session for account1? (sounds likely given the behavior you observed)
- does it initiate a session as account2, and discard the account1 session

either way it doesn't seem like a security issue with the auth flow, but does seem like it could result in a UX problem - my advice would be that on the reset password page, you clearly show the user the email address or username of the account that you are about to change the password for

hope this helps!

**Reply by tawjaw** [reply]
Thanks for the reply.

My concern with the security issue is based on how I thought reset password works. But I might be wrong.

What I understand is that the email for resetting the password gives a special access token that allows for updating the user password. Without it, a logged in user should not be able to update their password.

But I think from what I've observed is that it's just a magic link login with a parameter specifying recover.

Is that true? 

If so, is there a way to not allow a password update (even if a user is authenticated) unless they get an email first and click on the link?

**Reply by awalias** [reply]
> But I think from what I've observed is that it's just a magic link login with a parameter specifying recover.

correct

> is there a way to not allow a password update (even if a user is authenticated) unless they get an email first and click on the link?

not at the moment, however there is currently [work being done](https://github.com/supabase/gotrue/pull/309) to enforce that the user re-authenticates (by providing the "old" password) as part of setting a new password which might achieve the same goal
